src/aosm/azext_aosm/deploy/deploy_with_arm.py

```diff
@@ -126,15 +139,17 @@ def deploy_vnfd_from_bicep(
                 # required from config for the default bicep template produced from
                 # building the NFDV using this CLI
                 logger.debug("Create parameters for default NFDV template.")
-                parameters = self.construct_vnfd_parameters()
+                parameters = self.construct_nfd_parameters()
 
-            logger.debug(parameters)
+            logger.debug(
+                "Parameters used for NF definition bicep deployment: %s", parameters
+            )
 
             # Create or check required resources
-            deploy_manifest_template = not self.nfd_predeploy(definition_type=VNF)
+            deploy_manifest_template = not self.nfd_predeploy()
```

Comment:
We could now potentially end up in an odd situation. Say the artifact manifest partially deployed, so some exist and some don't, we will try to deploy again and I think this will fail because of AOSM bug/feature with artifact manifests?  I'm not actually sure what would happen.
I don't think the code wants to work in this odd situation, but it is worth a test of what happens, and if the user gets into a stuck situation where the CLI can't do it's job, add some error logging telling them how to workaround (delete all artifact manifests and start again)

Reply:
Sorry I've put this comment on the wrong line, should be 417.

Reply:
I think that is already covered in `do_config_artifact_manifests_exist()` ~line 402.